### PR TITLE
homework-10

### PR DESCRIPTION
**Pull request by Grygir**

### CODE REVIEW

File: homework-10/homework10.sql [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+  ('Middle'),
+  ('Senior');
+
+CREATE TABLE main.employees (
+  id SERIAL PRIMARY KEY,
+  first_name VARCHAR(255) NOT NULL,
+  last_name VARCHAR(255) NOT NULL,
+  position_id INT NOT NULL,
+  level_id INT NOT NULL,
+  salary NUMERIC(10, 2),
```
Comment (alinkin) [inline homework-10/homework10.sql:32]: should be `NOT NULL` too as per requirements

File: homework-10/homework10.sql [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  (4, 3),
+  (5, 3),
+  (14, 3);
+
+CREATE TABLE main.mentors (
+  mentor_id INT NOT NULL,
+  mentee_id INT NOT NULL CHECK ( mentee_id != mentor_id ) UNIQUE,
+  FOREIGN KEY (mentor_id) REFERENCES main.employees (id),
+  FOREIGN KEY (mentee_id) REFERENCES main.employees (id),
+  UNIQUE (mentor_id, mentee_id)
```
Comment (alinkin) [inline homework-10/homework10.sql:112]: makes sense to make it a primary key

File: homework-10/homework10.sql [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+-- Співробітник отримує підвищення та переводиться з проекту А на проект Б. Необхідно оновити його level,
+-- зарплату та проект, на якому він працює, знаючи його імʼя та прізвище
+BEGIN;
+
+UPDATE main.employees
+SET level_id = 2, salary = 1200
+WHERE first_name = 'Mattie' AND last_name = 'Wang';
+
+DELETE FROM main.employee_projects USING main.employees, main.projects
```
Comment (alinkin) [inline homework-10/homework10.sql:184]: this query will delete all rows in `main.employee_projects`.

I recommend using sub-SELECT queries instead

File: homework-10/homework10.sql (hunk 1 above)
Comment (Grygir) [reply homework-10/homework10.sql:32]: Interns usually work for free, that's why I left it nullable. Ok, will fix.

File: homework-10/homework10.sql (hunk 3 above)
Comment (Grygir) [reply homework-10/homework10.sql]: fixed

File: homework-10/homework10.sql (hunk 1 above)
Comment (alinkin) [reply homework-10/homework10.sql]: not in Sigma))

**Review by alinkin — APPROVED** [review]

**Pull request merged**